GUI/Util/DoubleTexBoxControl.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Windows.Controls;
+using System.Windows.Input;
+
+namespace MetaMorpheusGUI
+{
+    /// <summary>
+    /// This text box requires input text to be decimal only.
+    /// </summary>
+    public class DoubleTexBoxControl : TextBox
+    {
+        protected override void OnPreviewTextInput(TextCompositionEventArgs e)
+        {
+            foreach (var character in e.Text)
```

Comment:
why not use `double.TryParse()`?

Reply:
With this approach, it prevents invalid input in real time. You can't type any invalid characters into the box. With the TryParse approach, you could type anything in, but it would only be rejected when you clicked away from the box/when focus is lost.

Reply:
got it!